openssl-dynamic/src/main/c/ssl.c

```diff
@@ -678,21 +935,34 @@ TCN_IMPLEMENT_CALL(jint, SSL, getError)(TCN_STDARGS,
     return SSL_get_error(ssl_, ret);
 }
 
-// How much did SSL write into this BIO?
-TCN_IMPLEMENT_CALL(jint /* nbytes */, SSL, pendingWrittenBytesInBIO)(TCN_STDARGS,
-                                                                     jlong bio /* BIO * */) {
-    BIO *b = J2P(bio, BIO *);
+TCN_IMPLEMENT_CALL(jint /* nbytes */, SSL, bioCtrlPending)(TCN_STDARGS, jlong bio /* BIO* */) {
+    BIO* b = J2P(bio, BIO*);
 
     if (b == NULL) {
         tcn_ThrowException(e, "bio is null");
         return 0;
     }
 
-    UNREFERENCED_STDARGS;
-
     return BIO_ctrl_pending(b);
 }
 
+#ifdef OPENSSL_IS_BORINGSSL
```

Comment:
> not hooking into it seems pretty sound.

The goal of re-using these APIs is that other applications which may use this BIO implementation won't have to have access to my "special APIs". The risk with hooking into `BIO_ctrl` is I implement these values in an unexpected way which may result in "bad behavior" from someone use may use another BIO and attempts to plug this BIO in. I think its unlikely this code will be used in another context because it already requires some special APIs so I will just make new APIs for these methods too.

> But my comment was just pointing out that BIO_ctrl_wpending and BIO_wpending are the exact same thing.

I see ... here is some additional info from OpenSSL's code base:
https://github.com/openssl/openssl/blob/master/crypto/bio/bio_lib.c#L567
```c
/*
 * It is unfortunate to duplicate in functions what the BIO_(w)pending macros
 * do; but those macros have inappropriate return type, and for interfacing
 * from other programming languages, C macros aren't much of a help anyway.
 */
```